### PR TITLE
pull externals with vcpkg

### PR DESCRIPTION
**Pull request by jakoch**
This PR contains: 
- the vcpkg integration, which enables to build the project using vcpkg dependencies
   - the packages listed in `vcpkg.json` equal the removed packages from the externals folder
   - fixed some header includes
- the appveyor CI integration, which provides the build matrix for windows for the triplets  `x64-windows` and `x64-windows-static` 
   - https://ci.appveyor.com/project/jakoch/ttauri/history
   - build artifacts are continously available (in the artifacts tab of the current build job)
   - automatic release to Github Releases on tag push is prepared, but commented out
- lots of CMakeLists.txt changes, fixes and additions to get CI working, but mostly reordering, comments and cleanups 
- added ShowBuildProperties.cmake to show properties of individual build targets during cmake generation
- added `RelWithDebInfo` to project config in `CMakeSettings.json`

Requirements to get this working
- you'll need an installation of vcpkg
  - `git clone https://github.com/microsoft/vcpkg c:\tools\vcpkg`
  - `cd c:\tools\vcpkg`
  - `bootstrap-vcpkg.bat`
  - `vcpkg integrate install`
  - test: `vcpkg search catch2`

Referencing the issues: 
 - #34 (build with vcpkg) 
 - #35 (create ci workflow) 

Please review and comment on the code, so that we can get it into a state, where it is ready to be merged.

### CODE REVIEW

File: CMakeSettings.json [hunk 1]
```diff
@@ -1,210 +1,144 @@
 ﻿{
-    "configurations": [
+  "configurations": [
+    {
+      "name": "x64-MSVC-Analysis",
+      "generator": "Ninja",
+      "configurationType": "Debug",
+      "buildRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\build\\${name}",
+      "installRoot": "${env.USERPROFILE}\\CMakeBuilds\\${workspaceHash}\\install\\${name}",
+      "cmakeCommandArgs": "",
```
Comment (takev) [inline CMakeSettings.json:9]: "cmakeCommandArgs": "-DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_TARGET_TRIPLET=x64-windows-static -DBUILD_SHARED_LIBS=OFF"

As vcpkg when installing ttauri, calls cmake with both the tripplet and build_shared_libs. It seems okay to have to set both in the cmakesettings.json as well.

File: CMakeSettings.json [hunk 2]
```diff
@@ -1,210 +1,144 @@
...
+      "buildCommandArgs": "",
+      "ctestCommandArgs": "-C Debug",
+      "inheritEnvironments": [ "msvc_x64_x64" ],
+      "variables": [
```
Comment (takev) [inline CMakeSettings.json:13]: The triplet is now passed as a cmakeCommandArgs

File: CMakeSettings.json [hunk 3]
```diff
@@ -1,210 +1,144 @@
...
-                    "type": "STRING"
-                }
-            ]
-        },
+          "name": "VCPKG_TARGET_TRIPLET",
+          "value": "x64-windows",
+          "type": "STRING"
+        }
+      ],
+      "cmakeExecutable": "E:/tools/vcpkg/downloads/tools/cmake-3.19.2-windows/cmake-3.19.2-win32-x86/bin/cmake.exe"
```
[the remark this answers is not in the document]
Comment (jakoch) [reply CMakeSettings.json]: Ok. This only affects Visual Studio. The CI env has cmake on the environment path.

File: CMakeSettings.json (hunk 1 above)
Comment (jakoch) [reply CMakeSettings.json]: Ok, switching to static builds by default.

File: CMakeLists.txt [hunk 4]
```diff
@@ -67,36 +89,126 @@ else(APPLE)
...
+        "set VULKAN_SDK=\"C:\\VulkanSDK\\1.0.0.0\"\n")
+endif()
+find_package(Vulkan REQUIRED)
+
+#-------------------------------------------------------------------
+# Setup Build Targets
+#-------------------------------------------------------------------
+
+add_library(ttauri)
+add_library(ttauri::tauri ALIAS ttauri)
```
Comment (takev) [inline CMakeLists.txt:167]: What is this about?

Comment (jakoch) [reply CMakeLists.txt:167]: This sets up a CMake namespace for the library. This enables the user of your library to use namespaced imported targets.
Which is the [recommended way](https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html?highlight=namespaced) to export libraries.

- Basically to give the target a more structured name and stop to care about the read header file names.
- It also allows to group things under the namespace, e.g. one could introduce `ttauri::docs` or `ttauri::subcomponent`. 
For instance, the `date::date` library has a timezone subcomponent `date::date-tz`. 
- The real advantage is that you get better CMake debug output/diagnostics, when working with your library in another project via `target_link_libraries`.

**Pull request merged**